### DISCUSSION TITLE
Problems with bounce processing

### DISCUSSION THREAD

**Discussion by racke** [sympa-community/sympa#1636] [General]
Got a problem where a bounce ends up with "No address found in message Sympa::Message".  Sisimai  is able to detect the bounce correctly. I think it is rather a standard error message:

SMTP error from remote mail server after RCPT TO: ...
    550 5.1.1 ...
    Recipient address rejected: User unknown in relay recipient table

I read the code and it looks like Sympa requires a multipart/report MIME entity (_parse_dsn in Sympa::Spindle::ProcessBounce).

@ikedas, is that true?

**Comment by ikedas** [comment]
Sympa can basically only handle multipart/report DSNs (the other heuristics in _anabounce() are too old and no longer work).   I think Sisimai would be a bit smarter than Sympa.

Can you show us that error message?  And could you please submit a new issue?

**Comment by racke** [comment]
Is there a way to capture the incoming bounce message before Sympa processed the email?

Will add a new issue.

**Reply by racke** [reply]
> Is there a way to capture the incoming bounce message before Sympa processed the email?
> 

Self service: Adjust xxx-owner alias by injecting `tee /tmp/bounce.txt | `.

**Comment by racke** [comment]
The server is running Postfix, so we should be able to process these bounces:
```

swaks --from=racke@minivend.de --to=test345-nevairbe@anonym.de 
=== Trying mail01.anonym.de:25...
=== Connected to mail01.anonym.de.
<-  220 mail01.anonym.de ESMTP Postfix
 -> EHLO kebnekaise
<-  250-mail01.anonym.de
<-  250-PIPELINING
<-  250-SIZE 14336000
<-  250-VRFY
<-  250-ETRN
<-  250-STARTTLS
<-  250-ENHANCEDSTATUSCODES
<-  250-8BITMIME
<-  250-DSN
<-  250 SMTPUTF8
 -> MAIL FROM:<racke@minivend.de>
<-  250 2.1.0 Ok
 -> RCPT TO:<test345-nevairbe@anonym.de>
<** 550 5.1.1 <test345-nevairbe@anonym.de>: Recipient address rejected: User unknown in relay recipient table
 -> QUIT
<-  221 2.0.0 Bye
=== Connection closed with remote host.


```

**Comment by ikedas** [comment]
It's not the recipient's server (Postfix) that generates the bounce. In this case, since the recipient's server rejected the delivery before the DATA command succeeded, i.e. it has not yet received the message, it is not responsible for the delivery. It should be the sender's server that may generate the bounce, because it has already received the message and it is responsible for the delivery.
